### DISCUSSION TITLE
What is the proper way to connect to a remote Redis Cluster with Renovate

### DISCUSSION THREAD

**Discussion by jsrice7391** [renovatebot/renovate#9743] [Request Help]
I am onboarding my company into using renovate. There is a lot of interest in creating a remote Redis Cache for the dependency updates.

Currently, there is limited information on how Renovate wants to connect with Redis. I see that Renovate ships with the `redis` node module which is what I assume it uses to connect.

In the case of self-hosting, if users need to connect through a bastion or use TLS to connect to the cluster, what would be the recommended URL to give the `redisUrl` parameter?

If I dove into the Redis connection, would you all welcome a PR that allows the user to configure Redis like: 

```js
redisUrl :{
   host: ""
   password: ""
   port: ""
}
```

**Comment by viceice** [comment]
https://www.npmjs.com/package/redis#user-content-options-object-properties
![image](https://user-images.githubusercontent.com/1798109/116247905-9babd100-a76b-11eb-8a81-33e62f4ae299.png)

I think that url property can take any of those options.

**Reply by viceice** [reply]
* No cluster support: https://github.com/NodeRedis/node-redis/issues/574
* No sentinel support: https://github.com/NodeRedis/node-redis/issues/302


So you need some own kind of proxy infront of a redis cluster, `haproxy` can do this, eg: https://artifacthub.io/packages/helm/dandydev-charts/redis-ha
